docs/multiversion-multiinstance-smoke-test.md

```diff
@@ -0,0 +1,53 @@
+I am not sure a PR is necessary:
+
+## Setup
+
+1. Install ServiceControl version 2.
+1. Add 1 instance of ServiceControl v2 with MSMQ. It would help to use a name relating it to v3, e.g. `v2_SC`
+   - Configure a unique error queue name. It would help to use a name relating it to v2, e.g. `v2_error`
+   - Configure a unique audit queue name. It would help to use a name relating it to v2, e.g. `v2_audit`
+1. Install ServiceControl version 3
+1. Add 1 instance of ServiceControl v3 with MSMQ. It would help to use a name relating it to v3, e.g. `v3_SC`
+   - Configure a unique error queue name. It would help to use a name relating it to v3, e.g. `v3_error`
+   - Configure a unique audit queue name. It would help to use a name relating it to v3, e.g. `v3_audit`
+1. Download or checkout the [FaultTolerance](https://docs.particular.net/samples/faulttolerance/) sample.
+1. Edit the configuration of the sample project
+   - Reconfigure the endpoint to use the MSMQ transport
+   
+## Scenario 1   
+1. Configure ServiceControl v3 as Remote.
+1. Restart v3 SC
+1. Configure ServiceControl v2 as Master. 
+1. Restart v2 SC
+1. Edit the configuration of the sample project
+   - Configure the error queue to the unique error queue assigned to v2 instance of SC
+   - Configure auditing to the unique audit queue assigned to v3 instance of SC.
+1. Run the sample, sending error messages to the v2 instance of SC
+1. Set the sample to successfully process messages
+1. Connect ServiceInsight to the v2 instance of ServiceControl
+1. Retry one or more failed messages in ServiceInsight
+1. Confirm the message(s) successfully processed in the sample
+1. Confirm the message status is Resolved in ServiceInsight
```

Comment:
I am actually not sure this is correct, or if it just looks like an audit message now. I do not it will show initially as a Failed message, and then Retry Pending after a retry is issue. I just didn't have time to test it today.

Reply:
Correct. In SI it will appear as a regular audited message. I wonder if we should raise an issue to change this. It could be useful to know that a message succeeded after being retried. Then again, the same could be said of Immediate and Delayed Retries.

The real issue is whether the SC instance thinks the failed message is resolved. You might need to resort to querying the API directly to determine this.

Reply:
Interesting I was assuming we'd be using SP foe this but SI certainly works as well